geode-core/src/main/java/org/apache/geode/internal/cache/PersistentOplogSet.java

```diff
@@ -592,9 +592,13 @@ long newOplogEntryId() {
    *
    * @param minAvailableSpace the minimum amount of space we need in this directory.
    */
-  DirectoryHolder getNextDir(int minAvailableSpace, boolean checkForWarning) {
+  DirectoryHolder getNextDir(long minAvailableSpace, boolean checkForWarning) {
     DirectoryHolder dirHolder = null;
     DirectoryHolder selectedHolder = null;
+    if (minAvailableSpace < parent.getMaxOplogSizeInBytes()
```

Comment:
This change looks okay but I don't like us checking SET_IGNORE_PREALLOCATE here since this is a testing only flag.
I also don't like the old contract on this method passing in minAvailableSpace. This is basically always going to be a small number (MIN_DIR_SIZE or the size of one operation). So I think it is fair to say we only want to use a directory if it has room for one oplog (which is parent.getMaxOplogSizeInBytes()). Keep in mind that if we can't find a directory like that we then call this method again with checkForWarning set to false in which case it will just use the next directory (if compaction is enabled (which is the default)). If we end up with no directories with room for one oplog and compaction is disabled then it should fail with "disk is full" exception.

So I'd recommend that you get rid of the minAvailableSpace parameter and always use parent.getMaxOplogSizeInBytes() as the minAvailableSpace in this method.

But I'm also okay with you just keeping your current changes.

Reply:
I dont like checking that flag either, as you said we are changing our code due to tests... I think the code should be changed to get rid of that flag, but I think that could be done in other PR.
Also, I will write another ticket for the removal of `minAvailableSpace`: I tried removing it, and it breaks more than 30 integration tests, so I think it deserves its own PR.

Reply:
Is it really true that minAVailableSpace will always be smaller than the maxOplogSize? It looks like that param comes from the size of the value, which I think could in theory be bigger than the oplog.

Reply:
I think it can happen, this is why I want to separate the removal from this PR, and check if it can be done. I would bet I saw a test where a big operation was performed and it was split between several oplog files, so I would say in that case minAvailableSpace was bigger. But I want to double check.

Reply:
Doing it in another PR sounds like a good idea.

Reply:
This method with an op size is only used if we have already decided to create a new oplog. I think Dan is right that it is possible for the first record of an oplog to be bigger than the max oplog size. In that case I think we just write it out and allow the oplog file to be bigger than the configured max. Users can configure a smaller max oplog size and if they do this and have large values then that first record could exceed the max size. Hopefully we test this. I don't know if any logic in the oplog code to split an operation and write it to two oplogs. I know we have logic that says if the op can not fit in the current oplog file (based on its max size) then start a new oplog.